### PR TITLE
[SPARK-48850][DOCS][SS][SQL] Add documentation for new options added to State Data Source

### PR DESCRIPTION
**Pull request by eason-yuchen-liu**
### What changes were proposed in this pull request?

In https://github.com/apache/spark/pull/46944 and https://github.com/apache/spark/pull/47188, we introduced some new options to the State Data Source. This PR aims to explain these new features in the documentation.

### Why are the changes needed?

It is necessary to reflect the latest change in the documentation website.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

The API Doc website can be rendered correctly.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: docs/structured-streaming-state-data-source.md [hunk 1]
```diff
@@ -144,16 +143,126 @@ The following configurations are optional:
...
+  <td>snapshotPartitionId</td>
+  <td>numeric value</td>
+  <td></td>
+  <td>If specified, only this specific partition will be read. Note that partition ID starts with 0. This option must be used together with 'snapshotStartBatchId'.</td>
+</tr>
+<tr>
+  <td>readChangeFeed</td>
+  <td>boolean</td>
+  <td>false</td>
+  <td>If set to true, will read the change of state over microbatches. The output table schema will also change. Two columns 'batch_id'(long) and 'change_type'(string) will be appended to the front. Option 'changeStartBatchId' must be specified with this option. Option 'batchId', 'joinSide', 'snapshotStartBatchId', 'snapshotPartitionId' is conflict with this option. An example usage of this option can be found below.</td>
```
Comment (HeartSaVioR) [inline docs/structured-streaming-state-data-source.md:162]: > The output table schema will also change. Two columns 'batch_id'(long) and 'change_type'(string) will be appended to the front.

We could simply defer to the next section, to make the explanation to be concise.

> is conflict with this option

cannot be used with this option (probably more clearer)

> An example usage of this option can be found below.

probably better to explicitly mention the section name? link would be even better.

**Review by HeartSaVioR — COMMENTED** [review]
Would you mind building the doc and capturing snapshot for the change? It would be much easier to review the change with markdown/code. Thanks!

**Comment by eason-yuchen-liu** [comment]
Screenshot 2024-07-11 at 11 43 28 AM
Screenshot 2024-07-11 at 11 43 42 AM

**Comment by eason-yuchen-liu** [comment]
CI failure is not relevant.

**Review by HeartSaVioR — APPROVED** [review]

**Comment by HeartSaVioR** [comment]
https://github.com/eason-yuchen-liu/spark/runs/27338180115

Looks like it only failed with protobuf breaking change. Not sure why, probably it tried to find the proto file from changeset? I'll ignore it anyway.

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Pull request closed without merging**